diff --git a/src/Mod/Material/App/MaterialValue.cpp b/src/Mod/Material/App/MaterialValue.cpp
--- a/src/Mod/Material/App/MaterialValue.cpp
+++ b/src/Mod/Material/App/MaterialValue.cpp
@@ -304,8 +304,13 @@ QString MaterialValue::getYAMLStringMultiLine() const
 {
     QString yaml;
     yaml = QStringLiteral(" |2");
+#if QT_VERSION < QT_VERSION_CHECK(5, 14, 0)
+    auto list = getValue().toString().split(QRegularExpression(QStringLiteral("[\r\n]")),
+                                            QString::SkipEmptyParts);
+#else
     auto list =
         getValue().toString().split(QRegularExpression(QStringLiteral("[\r\n]")), Qt::SkipEmptyParts);
+#endif
     for (auto& it : list) {
         yaml += QStringLiteral("\n      ") + it;
     }
