diff --git a/src/Base/CMakeLists.txt b/src/Base/CMakeLists.txt
--- a/src/Base/CMakeLists.txt
+++ b/src/Base/CMakeLists.txt
@@ -126,10 +126,12 @@ endif(SWIG_FOUND)
 SOURCE_GROUP("pycxx" FILES ${PYCXX_SOURCES})
 
-# auto-generate resource file with all available translations
-set (FreeCADBase_TR_QRC ${CMAKE_CURRENT_BINARY_DIR}/Resources/Base_translation.qrc)
-qt_find_and_add_translation(QM_SRCS "Resources/translations/*_*.ts"
-    ${CMAKE_CURRENT_BINARY_DIR}/Resources/translations)
-qt_create_resource_file(${FreeCADBase_TR_QRC} ${QM_SRCS})
-qt_add_resources(FreeCADBase_QRC_SRCS ${FreeCADBase_TR_QRC})
+if(BUILD_GUI)
+    # auto-generate resource file with all available translations
+    set (FreeCADBase_TR_QRC ${CMAKE_CURRENT_BINARY_DIR}/Resources/Base_translation.qrc)
+    qt_find_and_add_translation(QM_SRCS "Resources/translations/*_*.ts"
+        ${CMAKE_CURRENT_BINARY_DIR}/Resources/translations)
+    qt_create_resource_file(${FreeCADBase_TR_QRC} ${QM_SRCS})
+    qt_add_resources(FreeCADBase_QRC_SRCS ${FreeCADBase_TR_QRC})
+endif()
 
 SET(FreeCADBase_Pyi_SRCS
