diff --git a/cMake/FreeCadMacros.cmake b/cMake/FreeCadMacros.cmake
--- a/cMake/FreeCadMacros.cmake
+++ b/cMake/FreeCadMacros.cmake
@@ -241,7 +241,14 @@ MACRO(SET_BIN_DIR ProjectName OutputName)
     if(WIN32)
         set_target_properties(${ProjectName} PROPERTIES DEBUG_OUTPUT_NAME ${OutputName}_d)
     else(WIN32)
-        # FreeCADBase, SMDS, Driver and MEFISTO2 libs don't depend on parts from CMAKE_INSTALL_LIBDIR
+        if(CMAKE_SYSTEM_NAME STREQUAL "OHOS")
+            get_target_property(_freecad_target_type ${ProjectName} TYPE)
+            if(_freecad_target_type STREQUAL "EXECUTABLE")
+                set_property(TARGET ${ProjectName} PROPERTY INSTALL_RPATH "\$ORIGIN/../lib")
+            else()
+                set_property(TARGET ${ProjectName} PROPERTY INSTALL_RPATH "\$ORIGIN")
+            endif()
+        # FreeCADBase, SMDS, Driver and MEFISTO2 libs don't depend on parts from CMAKE_INSTALL_LIBDIR
-        if(NOT ${ProjectName} MATCHES "^(FreeCADBase|SMDS|Driver|MEFISTO2)$")
+        elseif(NOT ${ProjectName} MATCHES "^(FreeCADBase|SMDS|Driver|MEFISTO2)$")
             if(${ARGC} STREQUAL 4)
                 set_property(TARGET ${ProjectName} APPEND PROPERTY INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/${ARGV3})
