--- a/src/gl/glstate.h
+++ b/src/gl/glstate.h
@@ -106,6 +106,9 @@ struct glstate_s {
     GLsizei             scratch_vertex_size;
     GLuint              scratch_indices;
     GLsizei             scratch_indices_size;
+#if defined(__OHOS__)
+    unsigned int        ohos_pending_draw_count;
+#endif
     // Implementation read
     GLenum              readf; // implementation Read Format
     GLenum              readt; // implementation Read Type
